server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -144,84 +143,42 @@ export class MetadataExtractionProcessor {
   async extractExifInfo(job: Job<IExifExtractionProcessor>) {
     try {
       const { asset, fileName }: { asset: AssetEntity; fileName: string } = job.data;
-      const exifData = await exifr.parse(asset.originalPath, {
-        tiff: true,
-        ifd0: true as any,
-        ifd1: true,
-        exif: true,
-        gps: true,
-        interop: true,
-        xmp: true,
-        icc: true,
-        iptc: true,
-        jfif: true,
-        ihdr: true,
+      const exifData = await exiftool.read(asset.originalPath).catch(e => {
+        this.logger.debug(`The exifData parsing failed due to: ${e}`);
+        throw new Error(`Can not parse exif data from file ${asset.originalPath}`);
       });
 
-      if (!exifData) {
-        throw new Error(`can not parse exif data from file ${asset.originalPath}`);
-      }
-
-      const createdAt = new Date(exifData.DateTimeOriginal || exifData.CreateDate || new Date(asset.createdAt));
+      const exifToDate = (exifDate: string | ExifDateTime | undefined) =>
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        exifDate ? new Date(exifDate.toString()!) : null;
 
+      const createdAt = exifToDate(exifData.DateTimeOriginal ?? exifData.CreateDate ?? asset.createdAt);
+      const modifyDate = exifToDate(exifData.ModifyDate);
```

Comment:
Should we need to include `asset.modifiedAt` here when the EXIF data is not available?

Reply:
I think that would be beneficial so I have pushed changes to reflect that.